edit/provisioning.cattle.io.cluster/rke2.vue

```diff
@@ -962,13 +977,75 @@ export default {
 
     canRemoveKubeletRow(row, idx) {
       return idx !== 0;
-    }
+    },
+
+    async waitForResources(hasClusters) {
+      // An admin has access to all resource schemas and can exercise all(?) actions against individual resources and resource collections
+      // Non-admins however begin with a very small set of schemas and some only with POST rights. After they have access to a single cluster
+      // they gain the rights required to continue.
+      // These rights are not pushed to the client, so behold....
+
+      // We only need the special case for non-admins who don't have access to any clusters
+      const mgmtUsers = await this.$store.dispatch('management/findAll', { type: MANAGEMENT.USER });
+      const mgmtUser = mgmtUsers.filter(mu => mu.isCurrentUser);
+      const isAdmin = await mgmtUser.isAdmin;
+
+      if (!isAdmin && !hasClusters) {
+        // Before refreshing schemas / resource types via resetting the subscription ensure that the backend has caught up with the RBAC
+        // changes following the cluster creation
+        const haveNewSchemas = await this.waitForNewSchemaAccess();
+
+        if (!haveNewSchemas) {
+          throw new Error('Failed to fetch new schemas, cannot continue');
+        }
+
+        // Given we're about to rip out a lot of resources from underneath components ensure we hide them first to avoid undefined.x errors
+        Vue.set(this, 'saving', true);
+
+        // Refresh that schema/resource instance states
+        await this.$store.dispatch('loadManagement', true);
+
+        // loadManagement refetches everything else.... but not `provisioning.cattle.io.clusters`, so request again
+        await this.$store.dispatch('management/findAll', { type: CAPI.RANCHER_CLUSTER });
+      }
+
+      await this.value.waitForMgmt();
+    },
+
+    async waitForNewSchemaAccess() {
+      // We use the management node as a schema that a non-admin doesn't have access to before their first cluster... but will eventually
+      // Once we have this we know the backend will provide the correct responses for when we reset
+      const msg = 'Mgmt Node Schema';
+      let i = 0;
+
+      console.log('Starting wait for', msg); // eslint-disable-line no-console
+
+      while (i < 10) {
+        i++;
+
+        try {
+          const nodeSchema = await this.$store.dispatch('management/request', { url: `/v1/${ SCHEMA }/${ MANAGEMENT.NODE }` });
+
+          if (!!nodeSchema) {
+            console.log('Wait for', msg, 'done'); // eslint-disable-line no-console
+
+            return true;
+          }
+        } catch {}
+        console.log('Wait for', msg, 'not done yet'); // eslint-disable-line no-console
+        await new Promise(resolve => setTimeout(resolve, 2000));
+      }
+
+      console.log('Wait for', msg, 'timed out'); // eslint-disable-line no-console
+
+      return false;
+    },
```

Comment:
This is clearly based on [waitForTestFn](https://github.com/rancher/dashboard/blob/master/plugins/steve/resource-instance.js#L514), why not pull it out into a util and reuse instead of rewriting?

(Also with the separate timeout timer removed, this is now a potentially unbounded and could wait forever if the server is not responding...)

Reply:
I couldn't find a neat way to have an async function inside setInterval. Strictly speaking we don't necessarily have to wait for the request to succeed/fail and just check that eventually we have it. Will investigate

Reply:
Moved to #3720